### PR TITLE
assert() -> AWS_ASSERT(), removed assert includes

### PR DESCRIPTION
**Pull request by justinboswell**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: source/huffman_generator/generator.c [hunk 1]
```diff
@@ -321,8 +321,6 @@ int main(int argc, char *argv[]) {
         "\n"
         "#include <aws/compression/huffman.h>\n"
         "\n"
-        "#include <assert.h>\n"
```
Comment (graebm) [inline source/huffman_generator/generator.c]: Not even code hidden in strings is safe from Justin's wrath 🔥

Comment (justinboswell) [reply source/huffman_generator/generator.c]: it would have made clang-tidy mad immediately :P

Comment (ColdenCullen) [reply source/huffman_generator/generator.c]: Let me tell you how much fun it is making sure generated code passes clang-tidy/format...

Comment (tautschnig) [reply source/huffman_generator/generator.c]: `// clang-format off` ?

Comment (ColdenCullen) [reply source/huffman_generator/generator.c]: Unfortunately, no, it's not for the includes, but most of the file.

**Pull request merged**